src/components/AccountMenu.tsx

```diff
@@ -0,0 +1,163 @@
+import Box from '@mui/material/Box'
+import Avatar from '@mui/material/Avatar'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import Divider from '@mui/material/Divider'
+import IconButton from '@mui/material/IconButton'
+import Tooltip from '@mui/material/Tooltip'
+import CurrencyExchangeIcon from '@mui/icons-material/CurrencyExchangeTwoTone'
+import LogoutIcon from '@mui/icons-material/LogoutTwoTone'
+import FavoriteIcon from '@mui/icons-material/FavoriteTwoTone'
+import ShoppingBasketIcon from '@mui/icons-material/ShoppingBasketTwoTone'
+import ContrastIcon from '@mui/icons-material/ContrastTwoTone'
+import { Skeleton, Switch, Typography } from '@mui/material'
+import Link from 'next/link'
+import TransitionsModal from './TransitionModal'
+import SettingsIcon from '@mui/icons-material/SettingsTwoTone'
+import { useCallback, useState } from 'react'
+import dynamic from 'next/dynamic'
+import AccountSettings from './AccountSettings'
+
+const ExchangeModal = dynamic(() => import('./ExchangeModal'), {
+	ssr: false,
+	loading: () => (
+		<>
+			<Skeleton variant='text' height={150}>
+				<Typography component={'h3'} variant='h3'>
+					Caps&Cash Exchange
+				</Typography>
+			</Skeleton>
+			<Skeleton variant='text' sx={{ fontSize: '3rem' }} />
+			<Skeleton variant='rectangular' height={250} />
+		</>
+	),
+})
+
+export default function AccountMenu() {
+	const [modalOpen, setModalOpen] = useState(false)
+	const [contentId, setContentId] = useState<string | null>(null)
+
+	const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+	const open = Boolean(anchorEl)
+
+	const handleClick = useCallback((e: React.MouseEvent<HTMLElement>) => {
+		setAnchorEl(e.currentTarget)
+	}, [])
+
+	const handleClose = useCallback(() => {
+		setAnchorEl(null)
+	}, [])
+
+	const handleModalOpen = useCallback((id: string) => {
+		setModalOpen(true)
+		setContentId(id)
+	}, [])
+
+	const handleModalClose = useCallback(() => {
+		setModalOpen(false)
+		
+	}, [])
```

Comment:
Those useCallback are redundant. They do not depend on any external variables

Reply:
resolved in account-settings branch